ci_framework/roles/build_containers/tasks/main.yml

```diff
@@ -0,0 +1,73 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure directories are present
+  ansible.builtin.file:
+    path: "{{ cifmw_build_containers_basedir }}/{{ item }}"
+    state: directory
+  loop:
+    - tmp
+    - artifacts
+    - logs
+
+- name: Install tcib
+  ansible.builtin.import_tasks: install.yml
+
+- name: Generate container_build.sh script
+  ansible.builtin.template:
+    src: templates/build_containers.sh.j2
+    dest: "{{ cifmw_build_containers_basedir }}/artifacts/build_containers.sh"
+    mode: 0777
+    force: true
+
+- name: Run container_build.sh script
+  become: true
+  args:
+    chdir: "{{ cifmw_build_containers_basedir }}/artifacts"
+  ansible.builtin.shell: |
+    set -o pipefail && bash build_containers.sh 2>&1 {{ cifmw_build_containers_timestamper_cmd }} > {{ cifmw_build_containers_basedir }}/logs/build.log
+  environment:
+   PATH: /usr/local/bin:{{ ansible_env.PATH }}
+
+- name: Make sure authfile exists
+  when:
+    - cifmw_build_containers_authfile_path is defined
+    - cifmw_build_containers_push_containers | bool
+  block:
+    - name: Check for authfile
+      ansible.builtin.stat:
+        path: '{{ cifmw_build_containers_authfile_path }}'
+      register: authfile_exist
+
+    - name: Make sure autfile exists
+      ansible.builtin.assert:
+        that:
+          - authfile_exist.stat.exists | bool
+
+- name: Retrieve built images  # noqa risky-shell-pipe
+  ansible.builtin.shell: "podman images | grep {{ cifmw_build_containers_container_name_prefix }} | awk '{ print $1 }'"
+  register: built_images
+  become: true
+  changed_when: false
+
+- name: "Push image: {{ item }}"
+  become: true
+  ansible.builtin.command: >
+    buildah push --tls-verify=false {{ item }}:{{ image_tag }}
```

Comment:
we need to replace `image_tag` to `cifmw_build_containers_image_tag`. We are hitting following error [here](https://logserver.rdoproject.org/85/285/1726272dfa1369a6b5c1482ee854cb871ca968d0/github-check/cifmw-tcib/ab5ce17/job-output.txt)
```
TASK [build_containers : Push image: {{ item }} _raw_params=buildah push --tls-verify=false {{ item }}:{{ image_tag }}
2023-07-25 06:57:31.336897 | controller | ] ***
2023-07-25 06:57:31.336905 | controller | Tuesday 25 July 2023  06:57:31 -0400 (0:00:00.824)       0:19:58.497 **********
2023-07-25 06:57:31.379836 | controller | fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'image_tag' is undefined. 'image_tag' is undefined\n\nThe error appears to be in '/home/zuul/src/github.com/openstack-k8s-operators/ci-framework/ci_framework/roles/build_containers/tasks/main.yml': line 66, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: \"Push image: {{ item }}\"\n  ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes. Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}
```